C++/CMakeLists.txt

```diff
@@ -157,6 +157,11 @@ add_executable(shunting_yard
         test/algorithm/string/shunting_yard.cpp)
 target_link_libraries(shunting_yard test_runner)
 
+# Heap's Algorithm
+add_executable(heaps_algorithm
```

Comment:
At first I thought we might want to change this to just `heap` or `heaps`, but since that's also the name of a data structure, I think it's okay to leave this with the `_algorithm` suffix (for both function and file names) even though that's contrary to the naming conventions for the rest of the project.

Reply:
I agree, both because of ambiguity and because the algorithm is indeed named after B. R. Heap.